furusiya/model/maps/generators/map_generator.py

```diff
@@ -7,60 +9,48 @@
 
 
 def generate_monsters(area_map, num_monsters):
+    enemies = [
+        ('bushslime', config.data.enemies.bushslime, colors.desaturated_green, GameObject),
+        ('steelhawk', config.data.enemies.steelhawk, colors.light_blue, GameObject),
+        ('tigerslash', config.data.enemies.tigerslash, colors.orange, GameObject),
+        ('salamander', config.data.enemies.salamander, colors.red, Salamander)
+    ]
+    probabilities = [
+        45,
+        30,
+        25,
+        10
+    ]
+
     for i in range(num_monsters):
         # choose random spot for this monster
         x, y = area_map.get_random_walkable_tile()
-            
-        choice = Game.instance.random.randint(0, 100)
-        
-        if choice <= 55: 
-            name = 'bushslime'
-            data = config.data.enemies.bushslime
-            colour = colors.desaturated_green
-        elif choice <= 55 + 30:
-            name = 'steelhawk'
-            data = config.data.enemies.steelhawk
-            colour = colors.light_blue
-        else:  # 15
-            name = 'tigerslash'
-            data = config.data.enemies.tigerslash
-            colour = colors.orange
+        name, data, colour, cls = Game.instance.random.choices(enemies, weights=probabilities)[0]
 
-        monster = monster_factory.create_monster(data, x, y, colour, name)
+        monster = monster_factory.create_monster(data, x, y, colour, name, cls)
         area_map.entities.append(monster)
 
 
 def generate_items(area_map, num_items):

```

Comment:
It would be nice to pass the items and probabilities in as parameters so we can unit-test this code.